src/aus.h

```diff
@@ -0,0 +1,329 @@
+
+        void decrement()
+        {
+            return;
+        }
+
+        void start()
+        {
+            _start = std::chrono::system_clock::to_time_t(
+                std::chrono::system_clock::now());
+            _is_running = true;
+        }
+
+        void stop()
+        {
+            if (!_is_running)
+            {
+                return;
+            }
+            _is_running = false;
+            long current_time = get_current_time();
+            _total += (current_time - _start);
+            _start = current_time;
+        }
+
+
+
+    private:
+        long _start;
+        long _total;
+        bool _is_running;
+
+        time_t get_current_time() 
+        {
+            return std::chrono::system_clock::to_time_t(
+                std::chrono::system_clock::now());
+        }
+
+    };
+
+    class aus_counter : public aus_value
+    {
+    public:
+        aus_counter(long value = 0) : _counter(value), _total(0) { }
+        
+        long get()
+        {
+            return _counter;
+        }
+
+        void set(long value) {
+            _counter = value;
+        }
+
+        void increment()
+        {
+            _counter++;
+        }
+
+        void decrement()
+        {
+            _counter--;
+        }
+
+        void start() { return; }
+
+        void stop() { return; }
+
+
+    private:
+        long _counter;
+        long _total; 
+    };
+
+#ifdef BUILD_AUS
+
+    class aus_data
+    {
+
+    public:
+        aus_data()
+        {
+            _start_time = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+            _librealsense_version = RS2_API_VERSION_STR;
+            _os_name = get_os_name();
+            _platform_name = PLATFORM;
+
+        }
+
+        std::string get_os_name()
+        {
+            #ifdef _WIN32
+            return "Windows";
+            #else
+            #ifdef __APPLE__
+            return "Mac OS";
+            #else
+            #ifdef __linux__
+            return "Linux";
+            #else
+            return "Unknown";
+            #endif
+            #endif
+            #endif
+        }
+
+        
+        void set(std::string key, long value)
+        {
+            if (_mp.find(key) != _mp.end())
+            {
+                _mp[key]->set(value);
+            }
+            else
+            {//_mp[key] = std::make_unique<aus_counter>(new aus_counter(value));
+                _mp[key] = new aus_counter(value);
+            }
+
+        }
+
+        void increment(std::string key)
+        {
+            if (_mp.find(key) != _mp.end())
+            {
+                _mp[key]->increment();
+            }
+            else
+            {
+               // _mp[key] = std::make_unique<aus_counter>(new aus_counter(1));
+                _mp[key] = new aus_counter(1);
+            }
+        }
+
+        void decrement(std::string key)
+        {
+            if (_mp.find( key ) != _mp.end() && _mp[key] >0)
+            {
```

Comment:
If it can't find it it's an error right? exception?

Reply:
done